Improve benchmarking

vrugtehagel opened the pull request:
The benchmarks were quite simple, to the point they weren't a very accurate measure of how Vento compares to its peers. I've created a more elaborate template, that includes some basic logic and printing. This'll (hopefully) make compilation times and render times a bit more reflective of what they are in the wild.

I've restructured the benchmarks a bit to make it easier to add additional languages, and added JSX, Pug and EJS to the list.

There are now 3 benchmarks; one for compilation speed (this does not render anything), one for only rendering (excluding compilation) and one for the combination. "Edge" has been excluded from the compilation-only and render-only benchmarks, because it doesn't seem to have a straight-forward API that would let us do these things separately.

To add a benchmark, add a template within `bench/templates/` and write a function in `bench/setup.ts` that returns a "render" function (i.e. a function that receives data and spits out a string).

If you're curious, JSX wins the compilation benchmark, Vento wins the render one, and Eta wins the combined one.

oscarotero commented:
Wow that's great!!

I think JSX is not reliable because Deno cache all imported modules (according to ESM specs), so once the first `await import(dir + "tmp.jsx")` is evaluated, the next calls just return the cached module.
To properly measure JSX, you can add a hash to ensure the filename is different for each import. For example:

```js
const { default: render } = await import(dir + "tmp.jsx" + `#${n++}`);
```
And instead of JSX, I'd mention the exact library used (React), because other libraries like Preact are much more faster.

Another thing: the tests are failing due the code format. Can you run `deno task fmt` to fix this?
Thanks!

vrugtehagel commented:
Great feedback, thanks! Good catch on the `import()` thing - I did think it seemed a bit _too_ fast, but didn't realize the caching in the `import()` statement was effectively ruining the JSX benchmark.

It no longer wins any of the benchmarks (the new winner for compilation is Eta, as you might've guessed) and it was scoring particularly poorly so I changed the library to Preact to try and give it a fighting chance. It's very slow (second-last) with its compilation now, which honestly surprises me a bit, but I guess it's making up for that in rendering speed (it's 3rd there).

After a bit of a ping pong match with the formatter, linter and type checker, it appears the build still fails due to missing permissions. I've added the most restrictive extra permissions to the `deno task bench` command, please have a look and see if you're okay with these added permissions.
